### ISSUE TITLE
Cannot login after 5 retries with RPC 100

### ISSUE THREAD

**Issue by vinnymac** [vinnymac/PokeNurse#211]
Placeholder for people with this issue

**Comment by YesThatAllen** [comment]
Here's what I'm seeing when I try to log in using PTC

`authenticate.js?6a8b:67 Error: Unsupported version for hashserver: 5704/1.27.04`

When running using `yarn run dev`I can see the full error, which is posted here

https://gist.github.com/YesThatAllen/93904f80051cb8a0234e13a9271525c2

EDIT - I'm using a PTC account, on macOS 10.12.13

**Comment by stayre** [comment]
Same here - both with and without hash. If it matters, I am on MacOS 10.12.3.

**Comment by lordjoshi** [comment]
I have the problem with Google account on Windows 7. No token. API is 5704.

**Comment by YesThatAllen** [comment]
I have two development computers.  On my laptop, I when I build from the develop branch, I get a failure.

Today I built from my desktop, and the develop branch works (PTC with a key, set to 5704)

**Comment by vinnymac** [comment]
A lot of variables, could be Niantic, could be the version of yarn building your dependency tree, could be the token service, etc. Not clear what it was in the above case.

**Comment by criddery** [comment]
Hello, thank you for the hard work in keeping this app going.  I love it.  I loaded the updated version, but I'm still getting the  "Fetching RPC endpoint failed, received status code 100".  I am attempting to log in using Google credentials and I see 2 new inputs (Token and API).  Seems API is defaulted in, but I do not know what to use for a Token.   Thank you - Colleen

**Comment by Redarion** [comment]
@criddery they are working on a fix on that thing. 
Anyway, the Token is an HashKey you can find on the net, even free or paid, but I suggest you to read more about what they are [here](https://talk.pogodev.org/d/55-api-hashing-service-f-a-q)

**Comment by truonghnguyen1234** [comment]
I purchased a HashKey and put it into the login, but I am still getting the failed notification. Is there another step that needs to be made?

**Comment by hacknug** [comment]
@truonghnguyen1234 You could try the `develop` branch where most issues have been fixed or wait for a release to come out *soon*.

**Comment by truonghnguyen1234** [comment]
@hacknug I'm not tech savvy enough to figure it out so I guess I'll just have to wait. :(

**Comment by vinnymac** [comment]
@truonghnguyen1234 shouldn't be long now. A release will be out this week for sure. We are just trying to iron out a few more details with evolve/transfer to make sure that users can actually use the number one features again.

**Comment by Redarion** [comment]
Just few more hours: next release is in progress!

**Comment by truonghnguyen1234** [comment]
@vinnymac So I downloaded the new version, and even with the key, the error comes up. Am I missing something?

**Comment by ghost** [comment]
Same issue still persists.

![image](https://cloud.githubusercontent.com/assets/16636556/24257370/0036d5ba-0feb-11e7-834c-b7476967c869.png)

**Comment by vinnymac** [comment]
@truonghnguyen1234 @therealabradolf well I didn't close this issue yet did I 😜

We are investigating if it is an issue with the production builds. It looks like @YesThatAllen has his builds working as they are supposed to, so I might just reupload his if I can confirm they are good. It looks like both mac and windows builds are having this issue right now.

**Comment by truonghnguyen1234** [comment]
@vinnymac Whoops! Got to excited seeing that update notification. Will hold on until official confirmation! Thank you!

**Comment by vinnymac** [comment]
@truonghnguyen1234 @therealabradolf I removed the old builds, I am reuploading ones that should work right now. Please give it a try whenever you would like 👍

**Comment by truonghnguyen1234** [comment]
@vinnymac No luck. :(

**Comment by YesThatAllen** [comment]
The most recent builds are working for me on a Mac. @truonghnguyen1234 let us know what OS isn't working?

**Comment by hacknug** [comment]
They work on both Windows 10 and macOS for me

**Comment by vinnymac** [comment]
@truonghnguyen1234 same error, same OS? Any differences at all? We could have you try some other builds too. If you hop on discord maybe we can debug this issue together.

**Comment by truonghnguyen1234** [comment]
@vinnymac Yeah, it looks like I'm still having an issue on my end. I'm wondering if I'm just missing a step when I purchased my hashkey. I'm on discord now as truonghnguyen1234 or add me on skype komododragon28@hotmail.com

**Comment by mhastrich** [comment]
Still not working for me on Windows 7 / Google Login / no Token.
I will try to Login with Token later.

**Comment by hacknug** [comment]
@mhastrich if by token you mean hashing key, there's no way you can log in without one. That's expected behaviour.

**Comment by mhastrich** [comment]
> @mhastrich if by token you mean hashing key, there's no way you can log in without one. That's expected behaviour.

The Login screen says a token is only needed for safer API interactions which means it's optional, doesn't it?

![2017_03_24_14_37_47_pokenurse_login](https://cloud.githubusercontent.com/assets/17780846/24296590/f27124de-109f-11e7-8824-075490a4ecab.png)

**Comment by vinnymac** [comment]
@mhastrich yea I mentioned this in the dev chat on discord yesterday.

We already fixed it on [develop](https://github.com/vinnymac/PokeNurse/blob/develop/app/screens/Login/components/LoginFormContainer.js#L156)

**Comment by Redarion** [comment]
> @mhastrich The Login screen says a token is only needed for safer API interactions which means it's optional, doesn't it?

actually now it is required.

**Comment by mhastrich** [comment]
Is the 150 RPM license enough if you only want to read the IV of your pokemon?

**Comment by hacknug** [comment]
It is way more than enough 👍

**Comment by Redarion** [comment]
If you can make more than 150 requests per minute by yourself i'll pay you the greater ones!
If you do not use any bot, it's more than enough! With pokenurse is almost impossibile so reach that number of requests.

**Comment by vinnymac** [comment]
It is certainly not impossible to go over 150 requests a minute but it is unlikely. You could at max do 15 evolves in a minute, and that would require praying to RNGesus. But if you were to simply spam the refresh button again and again, I am sure you could get higher than that.

**Comment by mhastrich** [comment]
I purchased a key but now i get "Error operation timed out after 4549 ms, 5 tries with error: Bad request to hashing server".
I left API open which should use default 5704 which is correct?

Edit: It's working at home now. Perhaps Firewall or it needed some seconds to activate!

**Comment by YesThatAllen** [comment]
I have an idea today's (April 8th) app & api update will require an update to pokenurse's dependencies.

**Comment by YesThatAllen** [comment]
In general, this issue is resolved by using API 6100.

There will be times when you can't get your key to work, it may be expired. See [Bad request to hashing server (expired key)](https://github.com/vinnymac/PokeNurse/issues/223)

A similar issue may come up with the next forced update, time will tell.

**Comment by stayre** [comment]
As I mentioned, My key had expired, so I bought a new one. The New key is the one with the problem.
